### PR TITLE
Answer:43 signal input 

### PR DESCRIPTION
**Pull request by stillst**

### CODE REVIEW

File: apps/angular/signal-input/src/app/user.component.ts [hunk 1]
```diff
@@ -19,23 +19,19 @@ const ageToCategory = (age: number): Category => {
...
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class UserComponent implements OnChanges {
-  @Input({ required: true }) name!: string;
-  @Input() lastName?: string;
-  @Input() age?: string;
+export class UserComponent {
+  $name = input.required<string>({ alias: 'name' });
+  $lastName = input('', { alias: 'lastName' });
+  $age = input(0, { alias: 'age', transform: Number });
```
Comment (tomalaforge) [inline apps/angular/signal-input/src/app/user.component.ts:33]: For transform, i didn't know that Number was working. 👍

File: apps/angular/signal-input/src/app/user.component.ts [hunk 2]
```diff
@@ -19,23 +19,19 @@ const ageToCategory = (age: number): Category => {
...
 
-  fullName = '';
-  category: Category = 'Junior';
-
-  ngOnChanges(): void {
-    this.fullName = `${this.name} ${this.lastName ?? ''}`;
-    this.category = ageToCategory(Number(this.age) ?? 0);
-  }
+  $fullName = computed(() => `${this.$name()} ${this.$lastName()}`);
+  $category = computed(() => ageToCategory(Number(this.$age())));
```
Comment (tomalaforge) [inline apps/angular/signal-input/src/app/user.component.ts:36]: why do you use Number again here, `age` should already be of type number

File: apps/angular/signal-input/src/app/user.component.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (stillst) [reply apps/angular/signal-input/src/app/user.component.ts:33]: I was just about to ask for your opinion on this syntax 😅
We've started using it at work to spot errors in templates like 
```
*ngIf="signalVariable"
```
At first glance, everything seems correct, but actually we need this
```
*ngIf="signalVariable()"
```

File: apps/angular/signal-input/src/app/user.component.ts (hunk 2 above)
Comment (stillst) [reply apps/angular/signal-input/src/app/user.component.ts]: you right, missed it

**Pull request closed without merging**